iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors.mlir

```diff
@@ -224,3 +224,31 @@ func @depthwise_conv2d(%input: tensor<1x113x113x96xf32>, %filter: tensor<3x3x96x
 // CHECK: scf.for
 // CHECK: scf.for
 // CHECK: linalg.depthwise_conv_2d_input_nhwc_filter_hwc
+
+func @subtensor_insert(%arg0: tensor<1x224x224x3xf32>) -> tensor<1x225x225x3xf32> {
+  %cst = constant 0.000000e+00 : f32
+  %0 = linalg.init_tensor [1, 225, 225, 3] : tensor<1x225x225x3xf32>
+  %1 = linalg.fill(%0, %cst) : tensor<1x225x225x3xf32>, f32 -> tensor<1x225x225x3xf32>
```

Comment:
this made me think of something Mahesh and I were chatting about a bit the other day: we should have some identification of DMA operations like fills and copies, as turning those into the corresponding flow ops will be significantly better (especially on GPUs). Maybe it'd be enough to have a canonicalizer on flow.dispatch.workgroups: if the body is linalg.fill -> tensor.store then turn that into a flow.tensor.splat, or if tensor.load -> tensor.store then turn into a flow.tensor.clone/etc.

Do either you or @MaheshRavishankar know if there's an issue tracking something like that, or should I file one?

Reply:
+1 on having such patterns to turn standalone fill/copy into corresponding DMA operations. Fill ops are used for initializing the output buffer and normally they will be fused together with the real producer ops and after a bunch of transformations those fill ops will be killed with vector store to load forwarding. Here it's a special case: the padding happened after has a different workload, so we cannot go down that route and have the fill "for free". Having DMA operations is better for such cases!

I don't recall we have such an issue to track; so please feel free to create one! (And I might get to it soon). :)

Reply:
Filed https://github.com/google/iree/issues/5105 to discuss/share examples/track!